packages/astro/src/vite-plugin-config-alias/index.ts

```diff
@@ -0,0 +1,99 @@
+  const config = tsr.tsconfigResolverSync({ cwd, searchName });
+
+  return config.exists ? config : null;
+};
+
+/** Returns a list of compiled aliases. */
+const getConfigAlias = (cwd: string | undefined): Alias[] | null => {
+  /** Closest tsconfig.json or jsconfig.json */
+  const config = getExistingConfig('tsconfig.json', cwd) || getExistingConfig('jsconfig.json', cwd);
+
+  // if no config was found, return null
+  if (!config) return null;
+
+  /** Compiler options from tsconfig.json or jsconfig.json. */
+  const compilerOptions = Object(config.config.compilerOptions);
+
+  // if no compilerOptions.baseUrl was defined, return null
+  if (!compilerOptions.baseUrl) return null;
+
+  // resolve the base url from the configuration file directory
+  const baseUrl = path.posix.resolve(path.posix.dirname(normalize(config.path)), normalize(compilerOptions.baseUrl));
+
+  /** List of compiled alias expressions. */
+  const aliases: Alias[] = [];
+
+  // compile any alias expressions and push them to the list
+  for (let [alias, values] of Object.entries(Object(compilerOptions.paths) as { [key: string]: string[] })) {
+    values = [].concat(values as never);
+
+    /** Regular Expression used to match a given path. */
+    const find = new RegExp(`^${[...alias].map((segment) => (segment === '*' ? '(.+)' : segment.replace(/[\\^$*+?.()|[\]{}]/, '\\$&'))).join('')}$`);
+
+    /** Internal index used to calculate the matching id in a replacement. */
+    let matchId = 0;
+
+    for (let value of values) {
+      /** String used to replace a matched path. */
+      const replacement = [...path.posix.resolve(baseUrl, value)].map((segment) => (segment === '*' ? `$${++matchId}` : segment === '$' ? '$$' : segment)).join('');
+
+      aliases.push({ find, replacement });
+    }
+  }
+
+  // compile the baseUrl expression and push it to the list
+  aliases.push({
+    find: /^(?!\.*\/)(.+)$/,
+    replacement: `${baseUrl
+      .split('')
+      .map((segment) => (segment === '$' ? '$$' : segment))
+      .join('')}/$1`,
+  });
+
+  return aliases;
+};
+
+/** Returns a Vite plugin used to alias pathes from tsconfig.json and jsconfig.json. */
+export default function configAliasVitePlugin(astroConfig: { projectRoot?: URL; [key: string]: unknown }): vite.PluginOption {
+  /** Aliases from the tsconfig.json or jsconfig.json configuration. */
+  const configAlias = getConfigAlias(astroConfig.projectRoot && url.fileURLToPath(astroConfig.projectRoot));
+
+  // if no config alias was found, bypass this plugin
+  if (!configAlias) return {} as vite.PluginOption;
+
+  return {
+    name: '@astrojs/vite-plugin-config-alias',
+    enforce: 'pre',
+    async resolveId(source: string, importer, options) {
+      /** Resolved source id from existing resolvers. */
+      const resolvedId = await this.resolve(source, importer, { skipSelf: true, ...options });
+
+      // if the existing resolvers find the file, return that resolution
```

Comment:
This gives priority to other resolvers (like the built-in Node resolution), correct? Might clarify in the comment.